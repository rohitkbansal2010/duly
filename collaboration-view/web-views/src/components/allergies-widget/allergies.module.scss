@import "~@styles/main.scss";
@import "~@styles/colors.scss";
@import "~@styles/typography.scss";
@import "~@styles/spacing-constants.scss";

.allergiesWidget {
  &Container {
    max-height: $widget-height;
    -ms-touch-action: pan-y;
    touch-action: pan-y;
  }

  &Content {
    position: relative;
    display: grid;
    grid-auto-rows: max-content;
    grid-template-columns: repeat(2, 1fr);
    -ms-touch-action: pan-y;
    touch-action: pan-y;

    &Item {
      margin-bottom: 1.25rem;

      &:nth-child(2n + 1) {
        padding-right: 0.625rem;
      }

      &:nth-child(2n) {
        padding-left: 0.625rem;
      }
    }

    &EmptyItem {
      margin-left: 0.625rem;
      background: lighten($gray, 7.5%);
      border-radius: 0.75rem;
    }
  }

  &Skeleton {
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    width: 100%;
    height: 100%;
    max-height: $widget-height;
  }
}

.noAllergies {
  &Container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    max-height: $widget-height;
    padding-bottom: 7.5rem;
    background: white;
    border-radius: 0.5rem;
  }

  &Icon {
    width: 14.5625rem;
    height: 14.5625rem;
    margin-bottom: 2rem;
    padding: 2.75rem;
    background: lighten($dupage-light-blue, 40%);
    border-radius: 100%;

    img {
      width: 100%;
      height: 100%;
    }
  }

  &Text {
    @include nexa-heavy-highlight;
    color: $dupage-light-blue;
    font-size: var(--font-size-40px-4px);
    line-height: var(--line-height-42px-4px);
  }
}
