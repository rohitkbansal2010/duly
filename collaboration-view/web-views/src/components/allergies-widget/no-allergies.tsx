import { noAllergiesIcon } from '@icons';

import styles from './allergies.module.scss';

export const NoAllergiesWidget = () => 
  (
    <div className={styles.noAllergiesContainer}>
      <div className={styles.noAllergiesIcon}>
        <img src={noAllergiesIcon} alt="no allergies" />
      </div>
      <div className={styles.noAllergiesText}>No Known Allergies</div>
    </div>
  );
