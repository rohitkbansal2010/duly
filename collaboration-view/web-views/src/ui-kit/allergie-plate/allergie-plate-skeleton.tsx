import { DulyLoader } from '@components/duly-loader';
import { ALLERGIES_LOADER_WIDTH } from '@constants';

import styles from './allergie-plate.module.scss';

export const AllergiePlateSkeleton = () => 
  (
    <div className={styles.allergiePlateSkeleton}>
      <DulyLoader width={ALLERGIES_LOADER_WIDTH} />
    </div>
  );
