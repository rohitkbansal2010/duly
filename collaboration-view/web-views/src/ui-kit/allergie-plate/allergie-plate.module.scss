@import "~@styles/colors.scss";
@import "~@styles/typography.scss";

.allergiePlate {
  &Container {
    display: flex;
    flex-direction: column;
    height: 100%;
    padding: 1.25rem 1.25rem 1.5rem 1.25rem;
    background: white;
    border-radius: 0.5rem;
  }

  &TopPart {
    display: flex;
    align-items: center;
    margin-bottom: 1.75rem;

    &Icon {
      position: relative;
      flex-shrink: 0;
      box-sizing: border-box;
      width: 3.75rem;
      height: 3.75rem;
      padding: 0.75rem;
      background: lighten($dupage-light-blue, 40%);
      border-radius: 100%;

      &Img {
        width: 100%;
        height: 100%;
      }

      &Warning {
        position: absolute;
        top: -0.15rem;
        right: -0.5rem;
        width: 1.5rem;
        height: 1.5rem;
      }
    }

    &Title {
      @include nexa-heavy-highlight;

      margin-left: 1.25rem;
      font-size: var(--font-size-24px-4px);
      line-height: 1.625rem;
      word-break: break-word;
    }
  }

  &BottomPart {
    height: 100%;
    padding: 1.65rem 1.65rem 0.65rem;
    background: lighten($gray, 10%);
    border: 0.0625rem solid $gray;
    border-radius: 0.75rem;

    &Container {
      display: flex;
      align-items: flex-start;
      justify-content: space-between;
    }

    &Label {
      position: relative;
      display: flex;
      align-items: flex-start;
      margin-bottom: 0.25rem;
      color: $dark-gray;
      font-weight: 700;
      font-size: var(--font-size-14px-4px);
      line-height: var(--line-height-16px-4px);
      text-transform: uppercase;
    }

    &IconWarning {
      width: 1.2rem;
      height: 1.2rem;
      margin-left: 0.125rem;
      transform: translate(0, -20%);
      img {
        width: 100%;
        height: 100%;
      }
    }

    &Value {
      display: flex;
      flex-direction: column;
      font-weight: bold;
      font-size: var(--font-size-16px-4px);
      line-height: var(--line-height-24px-4px);

      &SeverityLevel {
        display: inline-block;
        width: 0.75rem;
        height: 0.75rem;
        margin-left: 0.5rem;
        border-radius: 100%;

        &--mild {
          background: $dupage-green;
        }

        &--moderate {
          background: $dupage-saffron;
        }

        &--severe {
          background: $dupage-orange;
        }
      }

      span {
        width: 90%;
        overflow-wrap: break-word;
      }
    }

    &Item {
      width: 25%;
    }

    &ReactionContainer {
      width: 50%;

      & > div[class^="allergiePlateBottomPartItem"] {
        width: 100%;
      }

      &Item {
        display: flex;
        width: 100%;

        span {
          width: 50%;
          padding-right: 5%;
          overflow-wrap: break-word;
        }
      }
    }
  }

  &Skeleton {
    position: relative;
    width: calc(50% - 0.625rem);
    margin-bottom: 1.25rem;
    background: $white;
    border-radius: 0.5rem;

    &:nth-last-child(-n + 2) {
      margin-bottom: 0;
    }
  }
}
