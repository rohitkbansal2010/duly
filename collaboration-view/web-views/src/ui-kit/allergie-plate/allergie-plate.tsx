import capitalize from 'lodash/capitalize';
import sortBy from 'lodash/sortBy';
import { useMemo } from 'react';

import {
  environmentAllergieIcon,
  foodAllergieIcon,
  biologicAllergieIcon,
  medicalAllergieIcon,
  severeLevelWarning,
  noAllergiesIcon as commonAllergyIcon, otherAllergiesIcon
} from '@icons';
import { AllergieCategoryType, AllergiesData } from '@types';
import { formatMDYYYYDate } from '@utils';

import styles from './allergie-plate.module.scss';
import { BottomPartItem } from './bottom-part-item';

const IconsMap: {
  [K in AllergieCategoryType]: string;
} = {
  Food: foodAllergieIcon,
  Medication: medicalAllergieIcon,
  Biologic: biologicAllergieIcon,
  Environment: environmentAllergieIcon,
  Other: otherAllergiesIcon,
};

const getIcon = (categories: AllergieCategoryType[] = []): string => {
  if (categories.length > 1 || categories.length === 0) {
    return commonAllergyIcon;
  }
  return IconsMap[categories[0]];
};

type SeverityItemLabelProps = { isSevereAllergie: boolean };

const SeverityItemLabel = ({ isSevereAllergie }: SeverityItemLabelProps) =>
  (
    <>
      Severity
      {isSevereAllergie && (
      <img
        className={styles.allergiePlateBottomPartIconWarning}
        src={severeLevelWarning}
        alt="severe level warning"
      />
      )}
    </>
  );

export const AllergiePlate = ({
  title = '',
  recorded = '',
  categories = [],
  reactions = [],
}: AllergiesData) => {
  const iconSrc = getIcon(categories);

  const sortedCategories = useMemo(() =>
    categories.sort(), [ categories ]);

  const isSevereAllergy = useMemo(
    () =>
      reactions.some(({ severity }) =>
        severity === 'Severe'),
    [ reactions ]
  );

  const sortedReactions = useMemo(() =>
    sortBy(reactions, [ 'title' ]), [ reactions ]);

  return (
    <div
      className={styles.allergiePlateContainer}
      data-testid="allergie-plate-item"
    >
      <div className={styles.allergiePlateTopPart}>
        <div className={styles.allergiePlateTopPartIcon}>
          <img className={styles.allergiePlateTopPartIconImg} src={iconSrc} alt="allergie icon" />
          {isSevereAllergy && (
            <img
              className={styles.allergiePlateTopPartIconWarning}
              src={severeLevelWarning}
              alt="severe level warning"
            />
          )}
        </div>
        <div className={styles.allergiePlateTopPartTitle}>{capitalize(title)}</div>
      </div>
      <div className={styles.allergiePlateBottomPart}>
        <div className={styles.allergiePlateBottomPartContainer}>
          <BottomPartItem title={<span data-testid="allergie-plate-date-recorded-title">Date Recorded</span>}>
            <span data-testid="allergie-plate-date-recorded-value">{formatMDYYYYDate(recorded)}</span>
          </BottomPartItem>
          <BottomPartItem title={<span data-testid="allergie-plate-category-title">Category</span>}>
            {sortedCategories.map(category =>
            (
              <span data-testid="allergie-plate-category-item" data-testval={capitalize(category)} key={category}>{capitalize(category)}</span>
            ))}
          </BottomPartItem>
          <div className={styles.allergiePlateBottomPartReactionContainer}>
            <BottomPartItem
              title={(
                <div className={styles.allergiePlateBottomPartReactionContainerItem}>
                  <span data-testid="allergie-plate-severity-title">
                    <SeverityItemLabel isSevereAllergie={isSevereAllergy} />
                  </span>
                  <span data-testid="allergie-plate-reaction-title">Reaction</span>
                </div>
              )}
            >
              {sortedReactions.map(({ title, severity }) =>
              (
                <div
                  key={title + severity}
                  className={styles.allergiePlateBottomPartReactionContainerItem}
                >
                  <span data-testid="allergie-plate-severity-item" data-testval={capitalize(severity)}>{capitalize(severity)}</span>
                  <span data-testid="allergie-plate-reaction-item" data-testval={capitalize(title)}>{capitalize(title)}</span>
                </div>
              ))}
            </BottomPartItem>
          </div>
        </div>
      </div>
    </div>
  );
};
