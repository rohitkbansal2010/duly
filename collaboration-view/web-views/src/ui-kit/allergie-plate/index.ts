export * from './allergie-plate';
export * from './allergie-plate-skeleton';
