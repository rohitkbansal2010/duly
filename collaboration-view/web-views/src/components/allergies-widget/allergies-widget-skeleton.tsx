import { AllergiePlateSkeleton } from '@ui-kit/allergie-plate';

import styles from './allergies.module.scss';

const SKELETON_COUNT = 6;

export const AllergiesWidgetSkeleton = () =>
  (
    <div className={styles.allergiesWidgetSkeleton}>
      {[ ...new Array(SKELETON_COUNT) ].map((_, index) =>
        <AllergiePlateSkeleton key={index} />)}
    </div>
  ); 
