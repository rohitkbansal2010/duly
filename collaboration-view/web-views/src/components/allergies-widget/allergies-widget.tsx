import sortBy from 'lodash/sortBy';
import { useMemo } from 'react';
import { useSelector } from 'react-redux';

import { FadedScroll } from '@components/faded-scroll-2';
import { AppointmentModuleWidgetsAlias } from '@enums';
import { useWidgetSkeleton } from '@hooks';
import { RootState } from '@redux/reducers';
import { AllergiesData } from '@types';
import { AllergiePlate } from '@ui-kit/allergie-plate';

import { AllergiesWidgetSkeleton } from './allergies-widget-skeleton';
import styles from './allergies.module.scss';
import { NoAllergiesWidget } from './no-allergies';

export const AllergiesWidget = () => {
  const allergies = useSelector(({ OVERVIEW_WIDGETS }: RootState) =>
    OVERVIEW_WIDGETS.allergies);

  const isSkeletonShown = useWidgetSkeleton(AppointmentModuleWidgetsAlias.ALLERGIES);

  const sortedAllergies = useMemo(() =>
    sortBy(allergies, allergy =>
      new Date(allergy.recorded)).reverse(), [ allergies ]);

  if (isSkeletonShown) {
    return <AllergiesWidgetSkeleton/>;
  }

  if (!allergies?.length) {
    return <NoAllergiesWidget/>;
  }

  const isOddLength = allergies && allergies?.length % 2 !== 0;

  return (
    <FadedScroll height="100%" className={styles.allergiesWidgetContainer}>
      <div className={styles.allergiesWidgetContent}>
        {sortedAllergies.map((allergy: AllergiesData) =>
            (
              <div key={allergy.id} className={styles.allergiesWidgetContentItem}>
                <AllergiePlate {...allergy} />
              </div>
            ))}
        {isOddLength && (
        <div className={`${styles.allergiesWidgetContentEmptyItem} ${styles.allergiesWidgetContentItem}`} />
          )}
      </div>
    </FadedScroll>
  );
};
