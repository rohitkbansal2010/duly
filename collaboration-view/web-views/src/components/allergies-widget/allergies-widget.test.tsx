import { render, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import * as redux from 'react-redux';

import { allergiesMockData } from '@mock-data';
import { AllergiePlate } from '@ui-kit/allergie-plate';

import { AllergiesWidget } from './allergies-widget';
import { AllergiesWidgetSkeleton } from './allergies-widget-skeleton';
import { NoAllergiesWidget } from './no-allergies';

describe('AllergiesWidget', () => {
  let component: ReturnType<typeof mount | typeof render>;

  beforeEach(() => {
    jest
      .spyOn(redux, 'useSelector')
      .mockImplementation((selector: (state: unknown) => unknown) =>
        selector({ OVERVIEW_WIDGETS: { allergies: allergiesMockData } }));

    component = mount(<AllergiesWidget />);
  });

  it('component should match snapshot', () => {
    component = render(<AllergiesWidget />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('skeleton should match snapshot', () => {
    component = render(<AllergiesWidgetSkeleton />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('no allergies widget should match snapshot', () => {
    component = render(<NoAllergiesWidget />);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('component should contain right count of allergies', () => {
    expect(component.find(AllergiePlate)).toHaveLength(allergiesMockData.length);
  });

  it('component should render no allegries widget if there is no allergies', () => {
    jest.spyOn(redux, 'useSelector')
      .mockImplementation((selector: (state: unknown) => unknown) =>
        selector({ OVERVIEW_WIDGETS: { allergies: [] } }));

    component = mount(<AllergiesWidget />);

    expect(component.find(AllergiePlate)).toHaveLength(0);
    expect(component.find(NoAllergiesWidget)).toHaveLength(1);
  });
});
