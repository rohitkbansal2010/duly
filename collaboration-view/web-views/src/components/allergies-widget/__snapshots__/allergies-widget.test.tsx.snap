// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllergiesWidget component should match snapshot 1`] = `
<div
  class="scrollWrapper allergiesWidgetContainer"
  style="height:100%"
>
  <div
    class="topScrollFade top0 default"
  />
  <div
    class="scrollContent"
  >
    <div
      class="allergiesWidgetContent"
    >
      <div
        class="allergiesWidgetContentItem"
      >
        <div
          class="allergiePlateContainer"
          data-testid="allergie-plate-item"
        >
          <div
            class="allergiePlateTopPart"
          >
            <div
              class="allergiePlateTopPartIcon"
            >
              <img
                alt="allergie icon"
                class="allergiePlateTopPartIconImg"
                src="food-allergie-icon.svg"
              />
              <img
                alt="severe level warning"
                class="allergiePlateTopPartIconWarning"
                src="severe-level-warning.svg"
              />
            </div>
            <div
              class="allergiePlateTopPartTitle"
            >
              Peanuts
            </div>
          </div>
          <div
            class="allergiePlateBottomPart"
          >
            <div
              class="allergiePlateBottomPartContainer"
            >
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-title"
                  >
                    Date Recorded
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-value"
                  >
                    2/21/2022
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-category-title"
                  >
                    Category
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-category-item"
                    data-testval="Food"
                  >
                    Food
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartReactionContainer"
              >
                <div
                  class="allergiePlateBottomPartItem"
                >
                  <div
                    class="allergiePlateBottomPartLabel"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-title"
                      >
                        Severity
                        <img
                          alt="severe level warning"
                          class="allergiePlateBottomPartIconWarning"
                          src="severe-level-warning.svg"
                        />
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-title"
                      >
                        Reaction
                      </span>
                    </div>
                  </div>
                  <div
                    class="allergiePlateBottomPartValue"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Severe"
                      >
                        Severe
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Anaphylaxis"
                      >
                        Anaphylaxis
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="allergiesWidgetContentItem"
      >
        <div
          class="allergiePlateContainer"
          data-testid="allergie-plate-item"
        >
          <div
            class="allergiePlateTopPart"
          >
            <div
              class="allergiePlateTopPartIcon"
            >
              <img
                alt="allergie icon"
                class="allergiePlateTopPartIconImg"
                src="medical-allergie-icon.svg"
              />
            </div>
            <div
              class="allergiePlateTopPartTitle"
            >
              Penicillin
            </div>
          </div>
          <div
            class="allergiePlateBottomPart"
          >
            <div
              class="allergiePlateBottomPartContainer"
            >
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-title"
                  >
                    Date Recorded
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-value"
                  >
                    2/18/2022
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-category-title"
                  >
                    Category
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-category-item"
                    data-testval="Medication"
                  >
                    Medication
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartReactionContainer"
              >
                <div
                  class="allergiePlateBottomPartItem"
                >
                  <div
                    class="allergiePlateBottomPartLabel"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-title"
                      >
                        Severity
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-title"
                      >
                        Reaction
                      </span>
                    </div>
                  </div>
                  <div
                    class="allergiePlateBottomPartValue"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Moderate"
                      >
                        Moderate
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Itching"
                      >
                        Itching
                      </span>
                    </div>
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Mild"
                      >
                        Mild
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Rush"
                      >
                        Rush
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="allergiesWidgetContentItem"
      >
        <div
          class="allergiePlateContainer"
          data-testid="allergie-plate-item"
        >
          <div
            class="allergiePlateTopPart"
          >
            <div
              class="allergiePlateTopPartIcon"
            >
              <img
                alt="allergie icon"
                class="allergiePlateTopPartIconImg"
                src="biologic-allergie-icon.svg"
              />
            </div>
            <div
              class="allergiePlateTopPartTitle"
            >
              Latex
            </div>
          </div>
          <div
            class="allergiePlateBottomPart"
          >
            <div
              class="allergiePlateBottomPartContainer"
            >
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-title"
                  >
                    Date Recorded
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-value"
                  >
                    2/14/2022
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-category-title"
                  >
                    Category
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-category-item"
                    data-testval="Biologic"
                  >
                    Biologic
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartReactionContainer"
              >
                <div
                  class="allergiePlateBottomPartItem"
                >
                  <div
                    class="allergiePlateBottomPartLabel"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-title"
                      >
                        Severity
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-title"
                      >
                        Reaction
                      </span>
                    </div>
                  </div>
                  <div
                    class="allergiePlateBottomPartValue"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Mild"
                      >
                        Mild
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Rush"
                      >
                        Rush
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="allergiesWidgetContentItem"
      >
        <div
          class="allergiePlateContainer"
          data-testid="allergie-plate-item"
        >
          <div
            class="allergiePlateTopPart"
          >
            <div
              class="allergiePlateTopPartIcon"
            >
              <img
                alt="allergie icon"
                class="allergiePlateTopPartIconImg"
                src="environment-allergie-icon.svg"
              />
            </div>
            <div
              class="allergiePlateTopPartTitle"
            >
              Pollen
            </div>
          </div>
          <div
            class="allergiePlateBottomPart"
          >
            <div
              class="allergiePlateBottomPartContainer"
            >
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-title"
                  >
                    Date Recorded
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-value"
                  >
                    2/10/2022
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-category-title"
                  >
                    Category
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-category-item"
                    data-testval="Environment"
                  >
                    Environment
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartReactionContainer"
              >
                <div
                  class="allergiePlateBottomPartItem"
                >
                  <div
                    class="allergiePlateBottomPartLabel"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-title"
                      >
                        Severity
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-title"
                      >
                        Reaction
                      </span>
                    </div>
                  </div>
                  <div
                    class="allergiePlateBottomPartValue"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Mild"
                      >
                        Mild
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Congestion"
                      >
                        Congestion
                      </span>
                    </div>
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Mild"
                      >
                        Mild
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Rush"
                      >
                        Rush
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="allergiesWidgetContentItem"
      >
        <div
          class="allergiePlateContainer"
          data-testid="allergie-plate-item"
        >
          <div
            class="allergiePlateTopPart"
          >
            <div
              class="allergiePlateTopPartIcon"
            >
              <img
                alt="allergie icon"
                class="allergiePlateTopPartIconImg"
                src="no-allergies-icon.svg"
              />
            </div>
            <div
              class="allergiePlateTopPartTitle"
            >
              Latex
            </div>
          </div>
          <div
            class="allergiePlateBottomPart"
          >
            <div
              class="allergiePlateBottomPartContainer"
            >
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-title"
                  >
                    Date Recorded
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-date-recorded-value"
                  >
                    2/4/2022
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartItem"
              >
                <div
                  class="allergiePlateBottomPartLabel"
                >
                  <span
                    data-testid="allergie-plate-category-title"
                  >
                    Category
                  </span>
                </div>
                <div
                  class="allergiePlateBottomPartValue"
                >
                  <span
                    data-testid="allergie-plate-category-item"
                    data-testval="Biologic"
                  >
                    Biologic
                  </span>
                  <span
                    data-testid="allergie-plate-category-item"
                    data-testval="Food"
                  >
                    Food
                  </span>
                </div>
              </div>
              <div
                class="allergiePlateBottomPartReactionContainer"
              >
                <div
                  class="allergiePlateBottomPartItem"
                >
                  <div
                    class="allergiePlateBottomPartLabel"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-title"
                      >
                        Severity
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-title"
                      >
                        Reaction
                      </span>
                    </div>
                  </div>
                  <div
                    class="allergiePlateBottomPartValue"
                  >
                    <div
                      class="allergiePlateBottomPartReactionContainerItem"
                    >
                      <span
                        data-testid="allergie-plate-severity-item"
                        data-testval="Mild"
                      >
                        Mild
                      </span>
                      <span
                        data-testid="allergie-plate-reaction-item"
                        data-testval="Rush"
                      >
                        Rush
                      </span>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="allergiesWidgetContentEmptyItem allergiesWidgetContentItem"
      />
    </div>
  </div>
  <div
    class="bottomScrollFade bottom0 default"
  />
</div>
`;

exports[`AllergiesWidget no allergies widget should match snapshot 1`] = `
<div
  class="noAllergiesContainer"
>
  <div
    class="noAllergiesIcon"
  >
    <img
      alt="no allergies"
      src="no-allergies-icon.svg"
    />
  </div>
  <div
    class="noAllergiesText"
  >
    No Known Allergies
  </div>
</div>
`;

exports[`AllergiesWidget skeleton should match snapshot 1`] = `
<div
  class="allergiesWidgetSkeleton"
>
  <div
    class="allergiePlateSkeleton"
  >
    <div
      class="dulyLoaderWrapper"
      style="width:8rem"
    >
      <div
        aria-label="animation"
        role="button"
        style="width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none"
        tabindex="0"
        title=""
      />
    </div>
  </div>
  <div
    class="allergiePlateSkeleton"
  >
    <div
      class="dulyLoaderWrapper"
      style="width:8rem"
    >
      <div
        aria-label="animation"
        role="button"
        style="width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none"
        tabindex="0"
        title=""
      />
    </div>
  </div>
  <div
    class="allergiePlateSkeleton"
  >
    <div
      class="dulyLoaderWrapper"
      style="width:8rem"
    >
      <div
        aria-label="animation"
        role="button"
        style="width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none"
        tabindex="0"
        title=""
      />
    </div>
  </div>
  <div
    class="allergiePlateSkeleton"
  >
    <div
      class="dulyLoaderWrapper"
      style="width:8rem"
    >
      <div
        aria-label="animation"
        role="button"
        style="width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none"
        tabindex="0"
        title=""
      />
    </div>
  </div>
  <div
    class="allergiePlateSkeleton"
  >
    <div
      class="dulyLoaderWrapper"
      style="width:8rem"
    >
      <div
        aria-label="animation"
        role="button"
        style="width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none"
        tabindex="0"
        title=""
      />
    </div>
  </div>
  <div
    class="allergiePlateSkeleton"
  >
    <div
      class="dulyLoaderWrapper"
      style="width:8rem"
    >
      <div
        aria-label="animation"
        role="button"
        style="width:100%;height:100%;overflow:hidden;margin:0 auto;outline:none"
        tabindex="0"
        title=""
      />
    </div>
  </div>
</div>
`;
